### PR TITLE
initial porting of ignore rules

### PR DESCRIPTION
**Pull request by devdoshi**
## Why
Describe the motivation for the changes.

## What
What's changing? Anything of note to call out?

## Validation
* [ ] CI passes
* [ ] Verified in staging
* etc...

### CODE REVIEW

File: workspaces/cli-server/src/spectacle/index.ts [hunk 1]
```diff
@@ -230,9 +232,28 @@ export class LocalCliDiffService implements IOpticDiffService {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+interface ILocalCliConfigRepositoryDependencies {
+  opticYmlPath: string;
+  ignoreFilePath: string;
+}
+
 //@TODO: use IgnoreFileHelper (see on-demand.ts)
```
Comment (devdoshi) [inline workspaces/cli-server/src/spectacle/index.ts:240]: this comment can be removed now

File: workspaces/cli-server/src/spectacle/index.ts [hunk 2]
```diff
@@ -230,9 +232,28 @@ export class LocalCliDiffService implements IOpticDiffService {
...
+      dependencies.ignoreFilePath,
+      dependencies.opticYmlPath
+    );
+  }
+
+  async addIgnoreRule(rule: string): Promise<void> {
+    return await this.ignoreFileHelper.appendRule(rule);
+  }
+  async listIgnoreRules(): Promise<string[]> {
+    return await this.listIgnoreRules();
```
Comment (devdoshi) [inline workspaces/cli-server/src/spectacle/index.ts:255]: this looks like it infinite loops? should it be using this.ignoreFileHelper?

File: workspaces/spectacle/src/in-memory/index.ts [hunk 3]
```diff
@@ -163,7 +164,13 @@ export class InMemoryCapturesService implements IOpticCapturesService {
 ////////////////////////////////////////////////////////////////////////////////
 
 export class InMemoryConfigRepository implements IOpticConfigRepository {
-  ignoreRequests: string[] = [];
+  addIgnoreRule(rule: string): Promise<void> {
+    return Promise.resolve(undefined);
```
Comment (devdoshi) [inline workspaces/spectacle/src/in-memory/index.ts:168]: do you plan to make it work by maintaining an array here?

Comment (acunniffe) [reply workspaces/spectacle/src/in-memory/index.ts:168]: it doesn't really need to 'work' this is only for persistence, and the contract of demo/testing entry points is that if you refresh, it resets

**Comment by acunniffe** [comment]
verified this works end-end and saves ignore rules you add on the UI to the ignore file

File: workspaces/cli-shared/package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
     "ora": "^4.0.4",
     "proper-lockfile": "^4.1.1",
     "semver": "^7.3.4",
-    "shape-hash": "1.0.25",
+    "shape-hash": "1.0.6",
```
Comment (devdoshi) [inline workspaces/cli-shared/package.json:43]: why back to 1.0.6?

File: workspaces/ui-v2/src/optic-components/hooks/diffs/SharedDiffState.ts [hunk 5]
```diff
@@ -160,6 +161,16 @@ export const newSharedDiffMachine = (
               assign({
                 results: (ctx) => updateUrlResults(ctx),
               }),
+              (ctx, event) => {
+                configRepository
+                  .addIgnoreRule(event.rule)
+                  .then((i) => {
+                    debugger;
```
Comment (devdoshi) [inline workspaces/ui-v2/src/optic-components/hooks/diffs/SharedDiffState.ts:168]: should there be some awaiting here?

**Review by devdoshi — COMMENTED** [review]
haha i can't approve it since I opened it but looks fine, address my questions though

File: workspaces/cli-shared/package.json (hunk 4 above)
Comment (acunniffe) [reply workspaces/cli-shared/package.json:43]: wip. there will be a PR an insourced one inside the repo.

**Pull request merged**